File: frontend/src/app/shared/question-bank/QuestionBankSectionDto.ts
import { TestQuestionTemplate } from '../TestQuestionTemplate'

export class QuestionBankSectionDto {
  id: number
  name: string

  parentSectionId: number
  parentSection: QuestionBankSectionDto | null
  questionTemplates: TestQuestionTemplate[]

  public static NullSection(): QuestionBankSectionDto {
    let res = new QuestionBankSectionDto()
    res.name = 'Без секции'
    res.id = null
    return res
  }
}

File: backend/src2/SharedModels/DTOs/QuestionBankSectionDto.cs
using System.Collections.Generic;
using JetBrains.Annotations;

namespace SharedModels.DTOs
{
    public class QuestionBankSectionDto
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public int? ParentSectionId { get; set; }

        public ICollection<QuestionTemplateDto> QuestionTemplates { get; set; }

        [UsedImplicitly]
        public QuestionBankSectionDto()
        {
        }

        public QuestionBankSectionDto(int id, string name, int? parentSectionId, ICollection<QuestionTemplateDto> questionTemplates)
        {
            Id = id;
            Name = name;
            ParentSectionId = parentSectionId;
            QuestionTemplates = questionTemplates;
        }
    }
}